libqtile/config.py

```diff
@@ -818,6 +818,20 @@ def __repr__(self) -> str:
         )
 
 
+def convert_deprecated_list(vals: list[str], name: str) -> re.Pattern:
+    # make a regex with OR on word boundaries
+    regex_input = r"^({})$".format("|".join(map(re.escape, vals)))
+    logger.warning(
+        "Your Match with the %s property is using lists which are deprecated, replace Match(%s=%s) with Match(%s=re.compile(r\"%s\")) after importing the 're' module",
```

Comment:
```suggestion
        "Your Match with the %s property is using lists which are deprecated, replace Match(%s=%s) with Match(%s=re.compile(r\"^%s$\")) after importing the 're' module",
```
Just need the anchors here I think.

Reply:
ah, the anchors are already included in the regex_input, see: 
```
Your Match with the wm_class property is using lists which are deprecated, replace Match(wm_class=['foot', 'xterm']) with Match(wm_class=re.compile(r"^(foot|xterm)$")) after importing the 're' module
```

Reply:
Ah yes. I remember that's why I didn't make the suggestion before!

Reply:
Ah yes it's confusing at first